packages/design-system/src/components/primitives/arrow-focus.tsx

```diff
@@ -0,0 +1,162 @@
+import type { KeyboardEvent } from "react";
+
+const focusableSelector = (extra: string) =>
+  [
+    "input:not([type=hidden])",
+    "select",
+    "textarea",
+    "button",
+    "a[href]",
+    "area[href]",
+    "summary",
+    "iframe",
+    "object",
+    "embed",
+    "audio[controls]",
+    "video[controls]",
+    "[contenteditable]",
+    "[tabindex]",
+  ]
+    .map((selector) => `${selector}:not([disabled]):not([hidden])${extra}`)
+    .join(",");
+
+const keyToDirection: Record<
+  string,
+  ["vertical" | "horizontal", "+" | "-"] | undefined
+> = {
+  ArrowUp: ["vertical", "-"],
+  ArrowDown: ["vertical", "+"],
+  ArrowLeft: ["horizontal", "-"],
+  ArrowRight: ["horizontal", "+"],
+};
+
+const ROW_ATTRIBUTE = "data-arrow-focus-row";
+const COLUMN_ATTRIBUTE = "data-arrow-focus-column";
+
+/**
+ * Example: `<button {...setArrowFocusRow(1)}>`.
+ * The value can be any unique undetifier of the row
+ */
+export const setArrowFocusRow = (rowId: number | string) => ({
+  [ROW_ATTRIBUTE]: rowId,
+});
+
+/**
+ * Example: `<button {...setArrowFocusColumn(1)}>`.
+ * The value can be any unique undetifier of the column
+ */
+export const setArrowFocusColumn = (columnId: number | string) => ({
+  [COLUMN_ATTRIBUTE]: columnId,
+});
+
+type FocusableElements = { elements: Element[]; currentIndex: number };
+
+type Filter =
+  | "all"
+  | "within-row"
+  | "within-column"
+  | "first-per-row"
+  | "first-per-column";
+
+const getFocusable = (
+  filter: Filter,
+  root: Element,
+  current: Element
+): FocusableElements => {
+  if (
+    filter === "all" ||
+    filter === "within-row" ||
+    filter === "within-column"
+  ) {
+    let extraSelector = "";
+    if (filter !== "all") {
+      const attribute =
+        filter === "within-row" ? ROW_ATTRIBUTE : COLUMN_ATTRIBUTE;
+      extraSelector = `[${attribute}="${current.getAttribute(attribute)}"]`;
```

Comment:
Need to be commented on why we want the attribute to have the same value as current[attribute]?